### PR TITLE
Implementation of Metrics API operations

### PR DESCRIPTION
**Pull request by piosz**
ref https://github.com/kubernetes/kubernetes/issues/23376

Initial implementation - there is still a bunch of things to implement, added todos in the code.

<del> should be merged after #1151 </del>

cc @davidopp @fgrzadkowski

### CODE REVIEW

File: metrics/apis/metrics/handlers.go [hunk 1]
```diff
@@ -0,0 +1,187 @@
+// Copyright 2015 Google Inc. All Rights Reserved.
```
Comment (mwielgus) [inline metrics/apis/metrics/handlers.go]: 2016 :)

File: metrics/apis/metrics/handlers.go [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
```
Comment (mwielgus) [inline metrics/apis/metrics/handlers.go]: Please write why we had so crazy path.

File: metrics/apis/metrics/handlers.go (hunk 1 above)
Comment (piosz) [inline metrics/apis/metrics/handlers.go]: Done

File: metrics/apis/metrics/handlers.go (hunk 2 above)
Comment (piosz) [inline metrics/apis/metrics/handlers.go]: Done

**Comment by mwielgus** [comment]
LGTM

**Pull request merged**